packages/react/src/primitives/RadioGroupField/RadioGroupField.tsx

```diff
@@ -1,37 +1,43 @@
+import * as React from 'react';
 import classNames from 'classnames';
-import { useMemo } from 'react';
 
 import { RadioGroupContext, RadioGroupContextType } from './context';
 import { FieldErrorMessage, FieldDescription } from '../Field';
 import { Flex } from '../Flex';
 import { Label } from '../Label';
-import { RadioGroupFieldProps, Primitive } from '../types';
+import { RadioGroupFieldProps, PrimitiveWithForwardRef } from '../types';
 import { ComponentClassNames } from '../shared/constants';
 import { useStableId } from '../shared/utils';
 
 // Note: RadioGroupField doesn't extend the JSX.IntrinsicElements<'input'> types (instead extending 'typeof Flex')
 // because all rest props are passed to Flex container
-export const RadioGroupField: Primitive<RadioGroupFieldProps, typeof Flex> = ({
-  children,
-  className,
-  defaultValue,
-  descriptiveText,
-  errorMessage,
-  hasError = false,
-  id,
-  isDisabled,
-  isRequired,
-  isReadOnly,
-  label,
-  labelHidden = false,
-  onChange,
-  name,
-  size,
-  value,
-  ...rest
-}) => {
+const RadioGroupFieldInner: PrimitiveWithForwardRef<
+  RadioGroupFieldProps,
+  typeof Flex
+> = (
+  {
+    children,
+    className,
+    defaultValue,
+    descriptiveText,
+    errorMessage,
+    hasError = false,
+    id,
+    isDisabled,
+    isRequired,
+    isReadOnly,
+    label,
+    labelHidden = false,
+    onChange,
+    name,
+    size,
+    value,
+    ...rest
+  },
+  ref
+) => {
   const fieldId = useStableId(id);
-  const radioGroupContextValue: RadioGroupContextType = useMemo(
+  const radioGroupContextValue: RadioGroupContextType = React.useMemo(
```

Comment:
do we need to import the entire `React` package just to use `useMemo` and `forwardRef`? I'm ok with using this pattern on tests, but IMO components should only import what's going to be used.

Reply:
I prefer the explicitness of `React.useMemo` over `useMemo`. 🤷‍♂️ I think it makes it easier at a glance to know the function came from React. I also prefer the React namespace import over explicit imports because it means we don't have to continually update the import line whenever I use a new hook from React. I doubt if there is going to be bundle difference between the two import options, so it's mostly stylistic afaik.